src/main/webapp/ChapMap.js

```diff
@@ -270,9 +271,9 @@ class ChapMap {
     let permMarker = this.permMarkers_[markerId];
 
     if (!permMarker) {
-      this.makeNewPermMarker_(markerId, title, body, coords)
+      this.makeNewPermMarker_(markerId, title, body, color, coords)
     } else {
-      this.updatePermMarker_(permMarker, coords, title, body);
+      this.updatePermMarker_(permMarker, coords, title, body, color);
```

Comment:
Maybe for consistency's sake it might make sense to have the parameters in the same/similar order?

Reply:
Good point, it actually might be a easier if I just pass the JSON object as a parameter instead of all the separate parameters.